app/coffee/DocManager.coffee

```diff
@@ -51,7 +52,13 @@ module.exports = DocManager =
 					newDocLines: lines
 					rev: doc.rev
 				}, "updating doc lines"
-				MongoManager.updateDoc project_id, mongoPath, lines, (error) ->
+				async.series [
+					(cb)->
+						# project collection is still primary so that needs to be successful first
+						MongoManager.updateDoc project_id, mongoPath, lines, cb
+					(cb)->
+						MongoManager.upsertIntoDocCollection project_id, doc_id, lines, doc.rev, cb
```

Comment:
An error in the upsert will pass the error back to the client, even though the update has actually gone through in the project. Is this a problem with revisions not being updated in web, even though they have in the database? (I don't think so, but just something to be aware of)